test/framework/ownerreference_helpers.go

```diff
@@ -37,9 +37,17 @@ import (
 	clusterctlcluster "sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster"
 	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
 	addonsv1 "sigs.k8s.io/cluster-api/exp/addons/api/v1beta1"
+	expv1 "sigs.k8s.io/cluster-api/exp/api/v1beta1"
 	"sigs.k8s.io/cluster-api/util/patch"
 )
 
+// ValidateOwnerReferencesOnUpdate checks that expected owner references are in place, deletes them, and verifies that expect owner references are properly rebuilt.
+// TODO: Extend this test to include all ClusterClass templates when v1beta2 is released.
+func ValidateOwnerReferencesOnUpdate(proxy ClusterProxy, namespace string, assertFuncs ...map[string]func(reference []metav1.OwnerReference) error) {
+	// Check that the ownerReferences are as expected on the first iteration.
+	AssertOwnerReferences(namespace, proxy.GetKubeconfigPath(), assertFuncs...)
```

Comment:
Regarding test coverage:
* I went through all our CRDs and looks like we covered everything (with the expected exceptions of provider, the ipam types and the types for which we don't have old apiVersions (KubeadmControlPlaneTemplate & DockerClusterTemplate)
* Is my assumption correct that AssertOwnerReferences enforces that we assert all objects in the ownerGraph? Which is basically the same as all objects that clusterctl move would move?

Reply:
One additional thing. We could go through all usages of our ownerRef utils to double-check if we missed an ownerRef that we should keep up-to-date